src/harness/testcases/relinquishment_testcase.py

```diff
@@ -229,6 +229,106 @@ def test_WINFF_FT_S_RLQ_7(self):
+         'highFrequency': 3610000000.0
+    }
+    grant_3 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_3['cbsdId'] = cbsd_id_2
+    grant_3['operationParam']['operationFrequencyRange'] = {
+         'lowFrequency': 3610000000.0,
+         'highFrequency': 3620000000.0
+    }
+    grant_4 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_4['cbsdId'] = cbsd_id_4
+    grant_4['operationParam']['operationFrequencyRange'] = {
+         'lowFrequency': 3620000000.0,
+         'highFrequency': 3630000000.0
+    }
+    request = {'grantRequest': [grant_2, grant_3, grant_4]}
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(response[0]['cbsdId'], cbsd_id_2)
+    self.assertEqual(response[0]['response']['responseCode'], 0)
+    grant_id_2 = response[0]['grantId']
+    self.assertEqual(response[1]['cbsdId'], cbsd_id_2)
+    self.assertEqual(response[1]['response']['responseCode'], 0)
+    grant_id_3 = response[1]['grantId']
+    self.assertEqual(response[2]['cbsdId'], cbsd_id_4)
+    self.assertEqual(response[2]['response']['responseCode'], 0)
+    grant_id_4 = response[2]['grantId']
+    grant_id_1 = 'A nonexistent grant id'
+    del request, response
+
+    # Relinquish grant
+    request = {'relinquishmentRequest': [
+        {'cbsdId': cbsd_id_2, 'grantId': grant_id_3}]}
+    response = self._sas.Relinquishment(request)['relinquishmentResponse'][0]
+    # Check the relinquishment response
+    self.assertEqual(response['cbsdId'], cbsd_id_2)
+    self.assertEqual(response['grantId'], grant_id_3)
+    self.assertEqual(response['response']['responseCode'], 0)
+
+    # Relinquish the grants
+    request = {'relinquishmentRequest': [
+        {'cbsdId': cbsd_id_1, 'grantId': grant_id_2},
+        {'cbsdId': cbsd_id_2, 'grantId': grant_id_1},
+        {'cbsdId': cbsd_id_1, 'grantId': grant_id_1},
+        {'cbsdId': cbsd_id_2, 'grantId': grant_id_4},
+        {'cbsdId': cbsd_id_2, 'grantId': grant_id_3}]}
+    response = self._sas.Relinquishment(request)['relinquishmentResponse']
+    # Check the relinquishment response
+    self.assertFalse('cbsdId' in response[0])
+    self.assertEqual(response[0]['grantId'], grant_id_2)
+    self.assertIn(response[0]['response']['responseCode'], [103, 105])
+    self.assertEqual(response[1]['cbsdId'], cbsd_id_2)
+    self.assertFalse('grantId' in response[1])
+    self.assertEqual(response[1]['response']['responseCode'], 103)
+    self.assertFalse('cbsdId' in response[2])
+    self.assertFalse('grantId' in response[2])
+    self.assertIn(response[2]['response']['responseCode'], [103, 105])
+    self.assertEqual(response[3]['cbsdId'], cbsd_id_2)
+    self.assertEqual(response[3]['grantId'], grant_id_4)
+    self.assertEqual(response[3]['response']['responseCode'], 103)
+    self.assertEqual(response[4]['cbsdId'], cbsd_id_2)
+    self.assertFalse('grantId' in response[4])
```

Comment:
grand_id_3 is relinquished earlier so it's no longer a valid grant. TS says invalid grant id should not be included in the response. I have requested Masoud/Kambiz to make correct in the test plan.